### PR TITLE
RD-5301: Synchronized mechanism for scanning zip and git files

### PR DESCRIPTION
**Pull request by NorbertAmartus**
## Description

Issue was caused by not scanning `main.tf` file via processing mechanism for `.git` files
Solution for getting all the necessary file paths to scan has been unified across mechanisms responsible for processing `.git` and `.zip` files, to make it more robust and bulletproof 🎸 

## Screenshots / Videos

https://user-images.githubusercontent.com/89005877/179215616-de009811-7f7e-44b3-adad-1afb56a467d8.mp4

https://user-images.githubusercontent.com/89005877/179215646-f254b665-773e-493e-9be0-3b913215b44a.mp4

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

 

[Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/3158/)
2 case scenario had failed, but not due to the introduced modifications
From analyzing the video from [artefacts](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/3158/artifact/cloudify-stage/test/cypress/videos/widgets/), it seems like the issue is related to appearing `GettingStartedModal`, which is blocking the UI and therefor some click commands are failing.
The issue is appearing on 2 last test scenarios, so it may be related to the [before method](https://github.com/cloudify-cosmo/cloudify-stage/blob/dad9ec133fd5beef06bb1cc75c16bb63f8c8ad17/test/cypress/integration/widgets/blueprints_spec.ts#L812-L815) of one of the last `describe` blocks

### Image with `GettingStartedModal` blocking the view
image

As the issue is not related to my modifications, I consider this test execution a success 🚀 

## Documentation

N/A

### CODE REVIEW

**Comment by qooban** [comment]
I'll review this PR tomorrow.

**Review by PatrycjaKubiczek — APPROVED** [review]
tested UI, works as expected 🎉

File: backend/routes/Terraform.ts [hunk 1]
```diff
@@ -35,6 +35,7 @@ const templatePath = path.resolve(__dirname, '../templates/terraform');
 const template = fs.readFileSync(path.resolve(templatePath, 'blueprint.ejs'), 'utf8');
 // NOTE: The idea behind the code below has been described in more details here: https://serverfault.com/questions/544156/git-clone-fail-instead-of-prompting-for-credentials
 const disableGitAuthenticationPromptOption = '-c core.askPass=echo';
+const terraformFilesToScan = ['main.tf', 'outputs.tf', 'variables.tf'];
```
Comment (qooban) [inline backend/routes/Terraform.ts:38]: That's interesting that we parse only predefined set of files. I thought that we should parse all `.tf` files in the module.

No issues here, just a comment to raise attention. I'll check with @MichaelShn if that's desired behavior.

**Review by qooban — APPROVED** [review]
LGTM! 👍🏻

File: backend/routes/Terraform.ts (hunk 1 above)
Comment (NorbertAmartus) [reply backend/routes/Terraform.ts:38]: Ou, that's indeed an interesting behavior 😄 
In that case I'll merge PR, as currently mechanism is adjusted to look for specific file names, implementing mechanism of auto detecting all `.tf` files would require some refactoring and extra amount of work
(most probably it should be done as a part of a separate ticket)

Thanks for raising a flag and for adding me to the terraform channel 💪

**Pull request merged**